### DISCUSSION TITLE
Add, Remove & Update Items

### DISCUSSION THREAD

**Discussion by timothyjblevins** [Shopify/function-examples#581] [API Ideas]
I've been looking for a way to add, remove and update items server side. We have products that need to be added in the background based on if an item was personalized or for larger items, a shipping supplement. I have all the logic running frontend but it really should be server side. It's basically validation and updating in one function.

**Comment by andrewchen5678** [comment]
Needing the same thing for gift with purchase or buy certain combination of items, get another item automatically added to cart with discount use cases, I thought Shopify already supports this kind of basic functionality since it has been around for so long and it already supports Buy X get Y coupon discounts, but surprisingly item Y doesn't get automatically added to cart.
